data_management/opensearch_indexer/opensearch_indexer/text_extraction.py

```diff
@@ -76,5 +80,56 @@ def extract_text(file_stream: bytes, file_extension: str) -> str:
     ) as temp:
         temp.write(file_stream)
         temp.flush()
-        context = textract.process(temp.name)
-    return context.decode("utf-8")
+        file_path = temp.name
+
+        try:
+            context = textract.process(file_path)
+            return context.decode("utf-8")
+
+        except Exception as e:
+            logger.warning(f"Textract failed on {file_path}: {e}")
+
+            if file_extension not in CONVERT_MAP:
+                raise e
+            try:
+                logger.info("Attempting to convert via LibreOffice...")
+                converted_path = convert_failed_files(file_path)
+                logger.info(f"Converted to: {converted_path}")
+                text = textract.process(converted_path)
+                return text.decode("utf-8")
+            except Exception as convert_err:
+                logger.error(f"LibreOffice conversion failed: {convert_err}")
+                raise convert_err
+
+
+def convert_failed_files(input_path: str) -> str:
```

Comment:
thin it would be simpler to test / simplify concerns of this function by passing in the output_file_type as an argument, rather than needing to pass in a constant?